nodebuilder/header/constructors.go

```diff
@@ -99,16 +99,14 @@ func newInitStore[H libhead.Header[H]](
 	ds datastore.Batching,
 	ex libhead.Exchange[H],
 ) (libhead.Store[H], error) {
-	s, err := store.NewStore[H](ds, store.WithParams(cfg.Store))
-	if err != nil {
-		return nil, err
+	opts := []store.Option{store.WithParams(cfg.Store)}
+	if MetricsEnabled {
+		opts = append(opts, store.WithMetrics())
```

Comment:
Looks like we have two different patters for initialising metrics now. Do you know why?